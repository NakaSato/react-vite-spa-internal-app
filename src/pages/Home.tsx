import React from "react";
import { Link } from "react-router-dom";
import { useAuth } from "../contexts/AuthContext";

const Home: React.FC = () => {
  const { isAuthenticated, user } = useAuth();

  return (
    <div className="flex flex-col items-center justify-center min-h-screen px-4">
      <div className="max-w-4xl mx-auto text-center">
        <h1 className="text-4xl md:text-6xl font-bold text-gray-900 mb-6">
          Welcome to Solar Projects SPA
        </h1>

        {isAuthenticated ? (
          <div>
            <p className="text-xl text-gray-600 mb-8">
              Hello {user?.fullName}! Ready to manage your solar projects?
            </p>
            <Link
              to="/dashboard"
              className="inline-flex items-center px-6 py-3 border border-transparent text-base font-medium rounded-md text-white bg-blue-600 hover:bg-blue-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-blue-500"
            >
              Go to Dashboard
              <svg
                className="ml-2 -mr-1 w-5 h-5"
                fill="currentColor"
                viewBox="0 0 20 20"
              >
                <path
                  fillRule="evenodd"
                  d="M10.293 3.293a1 1 0 011.414 0l6 6a1 1 0 010 1.414l-6 6a1 1 0 01-1.414-1.414L14.586 11H3a1 1 0 110-2h11.586l-4.293-4.293a1 1 0 010-1.414z"
                  clipRule="evenodd"
                />
              </svg>
            </Link>
          </div>
        ) : (
          <div>
            <p className="text-xl text-gray-600 mb-8">
              A comprehensive platform for managing solar energy projects with
              real-time monitoring and advanced analytics.
            </p>

            <div className="space-y-4 sm:space-y-0 sm:space-x-4 sm:flex sm:justify-center">
              <Link
                to="/login"
                className="inline-flex items-center px-6 py-3 border border-transparent text-base font-medium rounded-md text-white bg-blue-600 hover:bg-blue-700 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-blue-500"
              >
                Sign In
                <svg
                  className="ml-2 -mr-1 w-5 h-5"
                  fill="currentColor"
                  viewBox="0 0 20 20"
                >
                  <path
                    fillRule="evenodd"
                    d="M3 3a1 1 0 011 1v12a1 1 0 11-2 0V4a1 1 0 011-1zm7.707 3.293a1 1 0 010 1.414L9.414 9H17a1 1 0 110 2H9.414l1.293 1.293a1 1 0 01-1.414 1.414l-3-3a1 1 0 010-1.414l3-3a1 1 0 011.414 0z"
                    clipRule="evenodd"
                  />
                </svg>
              </Link>

              <Link
                to="/about"
                className="inline-flex items-center px-6 py-3 border border-gray-300 text-base font-medium rounded-md text-gray-700 bg-white hover:bg-gray-50 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-blue-500"
              >
                Learn More
              </Link>
            </div>
          </div>
        )}

        {/* Features Section */}
        <div className="mt-16 grid grid-cols-1 gap-8 sm:grid-cols-3">
          <div className="text-center">
            <div className="text-4xl mb-4">‚òÄÔ∏è</div>
            <h3 className="text-lg font-medium text-gray-900 mb-2">
              Project Management
            </h3>
            <p className="text-gray-600">
              Efficiently manage solar installation projects from planning to
              completion.
            </p>
          </div>

          <div className="text-center">
            <div className="text-4xl mb-4">üìä</div>
            <h3 className="text-lg font-medium text-gray-900 mb-2">
              Real-time Analytics
            </h3>
            <p className="text-gray-600">
              Monitor energy production and system performance with advanced
              analytics.
            </p>
          </div>

          <div className="text-center">
            <div className="text-4xl mb-4">üë•</div>
            <h3 className="text-lg font-medium text-gray-900 mb-2">
              Team Collaboration
            </h3>
            <p className="text-gray-600">
              Collaborate with your team and manage roles and permissions
              seamlessly.
            </p>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Home;
