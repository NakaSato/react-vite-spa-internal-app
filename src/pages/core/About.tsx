import React from "react";

export default function About() {
  return (
    <div className="flex flex-col items-center justify-center h-screen">
      <h1 className="text-4xl font-bold">About Us</h1>
      <p className="mt-4 text-lg">
        This is the about page of your application.
      </p>
    </div>
  );
}
