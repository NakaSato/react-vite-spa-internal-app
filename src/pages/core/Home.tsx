import React from "react";
import { Link } from "react-router-dom";
import { useAuth } from "../shared/hooks/useAuth";

const Home: React.FC = () => {
  const { isAuthenticated, user } = useAuth();

  return (
    <div
      className="min-h-screen flex flex-col items-center justify-center px-4 bg-cover bg-center bg-no-repeat relative overflow-hidden"
      style={{
        backgroundImage: "url('/images/dashboard.jpg')",
      }}
    >
      {/* Enhanced Background Overlay with Gradient */}
      <div className="absolute inset-0 bg-gradient-to-br from-black/50 via-blue-900/30 to-black/60"></div>

      {/* Animated Background Elements */}
      <div className="absolute inset-0 overflow-hidden pointer-events-none">
        <div className="absolute top-1/4 left-1/4 w-64 h-64 bg-blue-500/10 rounded-full blur-3xl animate-pulse"></div>
        <div className="absolute bottom-1/4 right-1/4 w-96 h-96 bg-yellow-400/10 rounded-full blur-3xl animate-pulse delay-1000"></div>
        <div className="absolute top-1/2 left-1/2 transform -translate-x-1/2 -translate-y-1/2 w-48 h-48 bg-white/5 rounded-full blur-2xl animate-bounce"></div>
      </div>

      <div className="max-w-6xl mx-auto text-center relative z-10 animate-fade-in-up">
        {/* Logo Section */}
        <div className="mb-20 py-10">
          <div className="flex flex-col lg:flex-row justify-center items-center gap-8 lg:gap-16 xl:gap-20 mb-12 animate-slide-up">
            {/* PEA Logo */}
            <div className="group relative">
              <div className="absolute -inset-2 bg-gradient-to-r from-yellow-400 to-orange-500 rounded-full blur opacity-30 group-hover:opacity-60 transition duration-500"></div>
              <div className="relative w-40 h-40 md:w-48 md:h-48 lg:w-56 lg:h-56 xl:w-64 xl:h-64 bg-white rounded-full flex items-center justify-center shadow-2xl ring-4 ring-white/20 transition-all duration-500">
                <div className="w-4/5 h-4/5 relative overflow-hidden rounded-full">
                  <img
                    src="/images/pea_logo.png"
                    alt="Provincial Electricity Authority"
                    className="w-full h-full object-contain filter drop-shadow-lg"
                    loading="lazy"
                    onError={(e) => {
                      const target = e.target as HTMLImageElement;
                      target.style.display = "none";
                    }}
                  />
                </div>
                <div className="absolute inset-0 bg-gradient-to-tr from-transparent via-white/20 to-transparent opacity-0 transition-opacity duration-500 rounded-full"></div>
              </div>
            </div>

            {/* PWA Logo */}
            <div className="group relative">
              <div className="absolute -inset-2 bg-gradient-to-r from-blue-400 to-cyan-500 rounded-full blur opacity-30 group-hover:opacity-60 transition duration-500"></div>
              <div className="relative w-40 h-40 md:w-48 md:h-48 lg:w-56 lg:h-56 xl:w-64 xl:h-64 bg-white rounded-full flex items-center justify-center shadow-2xl ring-4 ring-white/20 transition-all duration-500">
                <div className="w-4/5 h-4/5 relative overflow-hidden rounded-full">
                  <img
                    src="/images/pwa_logo.png"
                    alt="PWA - Solar Energy Partner"
                    className="w-full h-full object-contain filter drop-shadow-lg transition-transform duration-500"
                    loading="lazy"
                    onError={(e) => {
                      const target = e.target as HTMLImageElement;
                      target.style.display = "none";
                    }}
                  />
                </div>
                <div className="absolute inset-0 bg-gradient-to-tr from-transparent via-white/20 to-transparent opacity-0 transition-opacity duration-500 rounded-full"></div>
              </div>
            </div>
          </div>

          {/* Partnership Badge */}
          <div className="flex justify-center animate-fade-in delay-300">
            <div className="relative group max-w-5xl">
              <div className="absolute -inset-1 bg-gradient-to-r from-white/20 to-blue-200/20 rounded-full blur opacity-50 group-hover:opacity-75 transition duration-500"></div>
              <div className="relative bg-white/10 backdrop-blur-md rounded-full px-8 py-4 md:px-12 md:py-5 lg:px-16 lg:py-6 xl:px-20 xl:py-7 border border-white/30 shadow-xl hover:shadow-2xl transition-all duration-300 hover:bg-white/15">
                <p className="text-white text-lg md:text-xl lg:text-2xl xl:text-3xl 2xl:text-4xl font-medium tracking-wide text-center leading-tight drop-shadow-lg">
                  โครงการจัดการพลังงานไฟฟ้าจากระบบผลิตไฟฟ้าพลังงานแสงอาทิตย์
                </p>
              </div>
            </div>
          </div>
        </div>

        {/* Content Section */}
        <div className="mb-12 space-y-6 animate-fade-in delay-500">
          {/* Subtitle */}
          <div className="relative">
            <p className="text-lg md:text-xl lg:text-2xl text-gray-200 drop-shadow-md font-light tracking-wide leading-relaxed">
              ระบบจัดการโครงการก่อสร้างภายในองค์กร
            </p>
            <p className="text-base md:text-lg text-blue-200 mt-2 drop-shadow font-medium">
              Internal Construction Management System
            </p>
          </div>
          {/* Call to Action */}
          <div className="flex flex-col sm:flex-row gap-6 justify-center items-center animate-fade-in delay-700">
            {isAuthenticated ? (
              <Link
                to="/dashboard"
                className="group relative inline-flex items-center px-10 py-4 text-lg font-semibold rounded-lg text-white bg-gradient-to-r from-slate-700 to-slate-800 hover:from-slate-800 hover:to-slate-900 focus:outline-none focus:ring-4 focus:ring-slate-500/50 transition-all duration-300 transform hover:scale-105 shadow-xl hover:shadow-2xl border border-slate-500/30"
              >
                <div className="absolute inset-0 bg-gradient-to-r from-white/5 to-transparent rounded-lg opacity-0 group-hover:opacity-100 transition-opacity duration-300"></div>
                <span className="relative flex items-center gap-3">
                  <svg
                    className="w-5 h-5"
                    fill="none"
                    stroke="currentColor"
                    viewBox="0 0 24 24"
                  >
                    <path
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      strokeWidth={2}
                      d="M19 21V5a2 2 0 00-2-2H7a2 2 0 00-2 2v16m14 0h2m-2 0h-4m-5 0H9m0 0H5m0 0h2M7 21V5a2 2 0 012-2h6a2 2 0 012 2v16"
                    />
                  </svg>
                  เข้าสู่ระบบจัดการโครงการ
                </span>
              </Link>
            ) : (
              <>
                <Link
                  to="/login"
                  className="group relative inline-flex items-center px-10 py-4 text-lg font-semibold rounded-lg text-white bg-gradient-to-r from-blue-700 to-blue-800 hover:from-blue-800 hover:to-blue-900 focus:outline-none focus:ring-4 focus:ring-blue-500/50 transition-all duration-300 transform hover:scale-105 shadow-xl hover:shadow-2xl border border-blue-500/30 hover:text-blue-200"
                >
                  <div className="absolute inset-0 bg-gradient-to-r from-white/5 to-transparent rounded-lg opacity-0 group-hover:opacity-100 transition-opacity duration-300"></div>
                  <span className="relative flex items-center gap-3">
                    <svg
                      className="w-5 h-5"
                      fill="none"
                      stroke="currentColor"
                      viewBox="0 0 24 24"
                    >
                      <path
                        strokeLinecap="round"
                        strokeLinejoin="round"
                        strokeWidth={2}
                        d="M11 16l-4-4m0 0l4-4m-4 4h14m-5 4v1a3 3 0 01-3 3H6a3 3 0 01-3-3V7a3 3 0 013-3h7a3 3 0 013 3v1"
                      />
                    </svg>
                    เข้าสู่ระบบ
                  </span>
                </Link>
              </>
            )}
          </div>
        </div>
      </div>
    </div>
  );
};

export default Home;
