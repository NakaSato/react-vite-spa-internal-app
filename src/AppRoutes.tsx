import React from "react";
import { Routes, Route, Navigate } from "react-router-dom";
import { useAuth } from "./hooks/useAuth";
import { Home, About, Login, Dashboard, Register } from "./pages";
import ApiExample from "./components/ApiExample";
import Navigation from "./components/Navigation";

const AppRoutes: React.FC = () => {
  const { isAuthenticated, isLoading } = useAuth();

  // Show loading spinner while checking authentication
  if (isLoading) {
    return (
      <div className="min-h-screen bg-gray-50 flex items-center justify-center">
        <div className="text-center">
          <div className="animate-spin rounded-full h-12 w-12 border-b-2 border-blue-600 mx-auto"></div>
          <p className="mt-4 text-gray-600">Loading...</p>
        </div>
      </div>
    );
  }

  return (
    <div className="App min-h-screen bg-gray-50">
      <Navigation />
      <main>
        <Routes>
          {/* Public Routes */}
          <Route path="/" element={<Home />} />
          <Route path="/about" element={<About />} />
          <Route path="/login" element={<Login />} />
          <Route path="/register" element={<Register />} />

          {/* Protected Routes */}
          <Route
            path="/dashboard"
            element={
              isAuthenticated ? <Dashboard /> : <Navigate to="/login" replace />
            }
          />
          <Route
            path="/api-example"
            element={
              isAuthenticated ? (
                <ApiExample />
              ) : (
                <Navigate to="/login" replace />
              )
            }
          />

          {/* Catch-all route */}
          <Route
            path="*"
            element={
              <Navigate to={isAuthenticated ? "/dashboard" : "/"} replace />
            }
          />
        </Routes>
      </main>
    </div>
  );
};

export default AppRoutes;
