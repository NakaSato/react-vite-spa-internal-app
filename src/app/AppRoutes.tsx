import React from "react";
import { Routes, Route, Navigate, useLocation } from "react-router-dom";
import { useAuth } from "../shared/hooks/useAuth";
import { DashboardProvider } from "../shared/contexts";
import { About, Login, Register, NotFound, ProjectDebug } from "../pages";
import SimpleHome from "../pages/core/SimpleHome";
import {
  DashboardLazy,
  ProjectDetailLazy,
  ProjectDetailRefactoredLazy,
  DailyReportsLazy,
  ProjectScheduleLazy,
  RealTimeProjectDashboardLazy,
} from "./LazyPages";
import { Navigation, RealTimeNotifications } from "../widgets";
import Footer from "../components/layout/Footer";
import ProtectedRoute from "../features/auth/ProtectedRoute";

const AppRoutesContent: React.FC = () => {
  const { isAuthenticated, isLoading } = useAuth();
  const location = useLocation();

  // Check if we're on the index page
  const isIndexPage = location.pathname === "/";

  // Show loading spinner while checking authentication
  if (isLoading) {
    return (
      <div className="min-h-screen bg-gray-50 flex items-center justify-center">
        <div className="text-center">
          <div className="animate-spin rounded-full h-12 w-12 border-b-2 border-blue-600 mx-auto"></div>
          <p className="mt-4 text-gray-600">Loading...</p>
        </div>
      </div>
    );
  }

  return (
    <div className="App min-h-screen bg-gray-50 flex flex-col">
      {!isIndexPage && <Navigation />}
      <main className="flex-grow">
        <Routes>
          {/* Public Routes */}
          <Route path="/" element={<SimpleHome />} />
          <Route path="/about" element={<About />} />
          <Route path="/login" element={<Login />} />
          <Route path="/register" element={<Register />} />

          {/* Protected Routes */}
          <Route
            path="/dashboard"
            element={
              <ProtectedRoute redirectToIndex={true}>
                <DashboardLazy />
              </ProtectedRoute>
            }
          />

          {/* Real-time Project Management Dashboard */}
          <Route
            path="/projects/realtime"
            element={
              <ProtectedRoute redirectToIndex={true}>
                <RealTimeProjectDashboardLazy />
              </ProtectedRoute>
            }
          />

          {/* Project Detail Page */}
          <Route
            path="/projects/:projectId"
            element={
              <ProtectedRoute redirectToIndex={true}>
                <ProjectDetailLazy />
              </ProtectedRoute>
            }
          />

          {/* Refactored Project Detail Page (for testing) */}
          <Route
            path="/projects/:projectId/refactored"
            element={
              <ProtectedRoute redirectToIndex={true}>
                <ProjectDetailRefactoredLazy />
              </ProtectedRoute>
            }
          />

          {/* Project Debug Page */}
          <Route
            path="/debug/projects"
            element={
              <ProtectedRoute redirectToIndex={true}>
                <ProjectDebug />
              </ProtectedRoute>
            }
          />

          {/* Project Schedule Management Page */}
          <Route
            path="/projects/:projectId/schedule"
            element={
              <ProtectedRoute redirectToIndex={true}>
                <ProjectScheduleLazy />
              </ProtectedRoute>
            }
          />

          {/* Daily Reports Management */}
          <Route
            path="/daily-reports"
            element={
              <ProtectedRoute redirectToIndex={true}>
                <DailyReportsLazy />
              </ProtectedRoute>
            }
          />
          {/* 404 Not Found - This will catch all unmatched routes */}
          <Route path="*" element={<NotFound />} />
        </Routes>
      </main>
      {!isIndexPage && <Footer />}
    </div>
  );
};

const AppRoutes: React.FC = () => {
  return (
    <DashboardProvider>
      <AppRoutesContent />
    </DashboardProvider>
  );
};

export default AppRoutes;
