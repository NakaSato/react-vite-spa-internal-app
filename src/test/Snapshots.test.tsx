import { describe, it, expect } from "vitest";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Home from "../pages/Home";
import About from "../pages/About";
import AppRoutes from "../app/AppRoutes";

describe("Snapshot Tests", () => {
  it("Home component matches snapshot", () => {
    const { container } = render(<Home />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("About component matches snapshot", () => {
    const { container } = render(<About />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("AppRoutes with home route matches snapshot", () => {
    const { container } = render(
      <MemoryRouter initialEntries={["/"]}>
        <AppRoutes />
      </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });

  it("AppRoutes with about route matches snapshot", () => {
    const { container } = render(
      <MemoryRouter initialEntries={["/about"]}>
        <AppRoutes />
      </MemoryRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
