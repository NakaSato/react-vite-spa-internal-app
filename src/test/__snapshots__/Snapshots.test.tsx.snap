// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Snapshot Tests > About component matches snapshot 1`] = `
<div
  class="flex flex-col items-center justify-center h-screen"
>
  <h1
    class="text-4xl font-bold"
  >
    About Us
  </h1>
  <p
    class="mt-4 text-lg"
  >
    This is the about page of your application.
  </p>
</div>
`;

exports[`Snapshot Tests > AppRoutes with about route matches snapshot 1`] = `
<div
  class="App"
>
  <div
    class="flex flex-col items-center justify-center h-screen"
  >
    <h1
      class="text-4xl font-bold"
    >
      About Us
    </h1>
    <p
      class="mt-4 text-lg"
    >
      This is the about page of your application.
    </p>
  </div>
</div>
`;

exports[`Snapshot Tests > AppRoutes with home route matches snapshot 1`] = `
<div
  class="App"
>
  <div
    class="flex flex-col items-center justify-center h-screen"
  >
    <h1
      class="text-4xl font-bold"
    >
      Welcome to the React Vite TypeScript SPA
    </h1>
    <p
      class="mt-4 text-lg"
    >
      This is the main page of your application.
    </p>
  </div>
</div>
`;

exports[`Snapshot Tests > Home component matches snapshot 1`] = `
<div
  class="flex flex-col items-center justify-center h-screen"
>
  <h1
    class="text-4xl font-bold"
  >
    Welcome to the React Vite TypeScript SPA
  </h1>
  <p
    class="mt-4 text-lg"
  >
    This is the main page of your application.
  </p>
</div>
`;
